clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -135,6 +136,18 @@
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
+    @Test
+    public void testMetricsReporterAutoGeneratedClientId() {
+        Properties props = new Properties();
+        props.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        props.setProperty(ConsumerConfig.METRIC_REPORTER_CLASSES_CONFIG, MockClientIdMetricsReporter.class.getName());
+        KafkaConsumer<String, String> consumer = new KafkaConsumer<>(
+                props, new StringDeserializer(), new StringDeserializer());
+
+        Assert.assertTrue(MockClientIdMetricsReporter.clientId.startsWith("consumer-"));
```

Comment:
Can you check that the auto-generated client id appears here?

Reply:
See comment above ^